src/Symfony/Component/Lock/LockableResourceTrait.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+trait LockableResourceTrait
+{
+    private ?Uuid $resourceIdentifier = null;
+
+    public function getResourceIdentifier(): string
+    {
+        return $this->resourceIdentifier ??= Uuid::v7();
```

Comment:
I don't think the dependency to UUID is worth it, let's find something else

Reply:
Thus, a random identifier is totally useless for locking: any other process trying to acquire a lock would generate a different resource identifier and so would try to acquire a different lock.

To me, this trait should be removed.

Reply:
Your code in the PR description will *always* succeed in acquiring the lock, unless you have a collision in the generation of the UUID (for a v7 UUID, this requires generating both UUIDs with the exact same timestamp and the same random data, which is basically impossible). So the equivalent code is removing all usages of the Lock component...

Reply:
You're right, I think I had a misunderstood about the internal mechanism of Lock. I'll update this accordingly!

Reply:
The interface is fine to add, but we cannot have a trait doing a magical implementation. The resource needs to be deterministic while embedded info about what domain considers the same resource regarding locking. So it must be implemented with domain knowledge.

Reply:
Thinking about this again regarding my own usages, I'm not even sure this interface is a good idea, because it means that each object will have a single lock resource associated with it.
For some of my objects, I have _different_ kinds of locks that can be taken for a given entity, so my lock resources are generated by combining the object id and a prefix identifying the kind of lock. I have a service that wraps the LockFactory and exposes several methods to acquire the different kind of locks, and such use case could not be refactored to use LockableResourceInterface (well, unless I use it on a special DTO that wraps the entity rather than the entity itself, but the benefit would be dubious).
So I'm not sure it makes sense to add that interface in core.
Note that your own project could implement such an interface on its own as the implementation is just `$lockFactory->createLock($lockableResource->getResourceIdentifier())`, which requires no change in the Lock component.

Reply:
Really makes sense. Thank you for this clear explanation!